sourcemash/api/items.py

```diff
@@ -17,18 +16,46 @@
     'category_1': fields.String,
     'category_2': fields.String,
     'summary': fields.String,
-    'category_1': fields.String,
-    'category_2': fields.String,
+    'totalVotes': fields.Integer,
     'uri': fields.Url('api.item')
 }
 
 
 class ItemAPI(Resource):
+    def __init__(self):
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('vote', type = int, required = True)
+        super(ItemAPI, self).__init__()
 
     def get(self, id):
         item = Item.query.get_or_404(id)
         return {'item': marshal(item, item_fields)}
 
+    @login_required
+    def put(self, id):
+        ''' Update item vote count '''
+        args = self.reqparse.parse_args()
+        item = Item.query.get_or_404(id)
+        
+        # Reject if size of vote is too large
```

Comment:
Extract these tests (abs(number) <= 1, requested vote != existing vote) to a form. Then call form validate on submit here. This will simplify the call here.

I extracted user_items out into a folder in sourcemash, so we can just add a forms.py to that folder now.